src/views/ObjectInfo.js

```diff
@@ -29,13 +29,14 @@ export default function ObjectInfo({ match }) {
   const [isNumberError, setIsNumberError] = useState(false);
 
   useEffect(() => {
-    // only fetch data for a potentially valid norad number
-    if (!isValidNumber(noradNumber)) {
-      setIsNumberError(true);
-    } else {
+    if (isValidNumber(noradNumber) && data.length === 0) {
       doFetch(`/object/info?norad_number=${noradNumber}`);
+    } else if (!isValidNumber(noradNumber)) {
```

Comment:
Couple points on this:
1. Norad Number is not valid if it equals 0
2. isValidNumber can use `^(\d{1,5})$` to check for a number with 1-5 digits.